Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSecurityEventFirewall.yaml

```diff
@@ -0,0 +1,303 @@
+            SrcDvcId = iff(isOutBound, RemoteMachineID, ""),
+            DstDvcId = iff(not(isOutBound), RemoteMachineID, ""),
+            SrcProcessId =  iff(isOutBound, tostring(ProcessId), ""),
+            DstProcessId =  iff(not(isOutBound), tostring(ProcessId), ""),
+            DstUserId = iff(isOutBound, RemoteUserID, ""),
+            SrcUserId = iff(not(isOutBound), RemoteUserID, ""),
+            DstHostname = iff(isOutBound, "", DvcHostname),
+            SrcHostname = iff(isOutBound, DvcHostname, "")
+          | project-away Application, RemoteMachineID, ProcessId, RemoteUserID
+            // *************** Postfilterring *****************************************************************
+          | where  (isnull(dstportnumber) or DstPortNumber == dstportnumber )
+          // *************** / Postfilterring *****************************************************************
+      };
+  // Main query -> outputs both schemas as one normalized table
+  WindowsFirewall_SecurityEvent  (starttime, endtime, srcipaddr_has_any_prefix, dstipaddr_has_any_prefix, ipaddr_has_any_prefix, dstportnumber, hostname_has_any, dvcaction, eventresult, disabled)
+      | extend 
+              DvcAction = iff(EventID in (5154, 5156, 5158), "Allow", "Deny"),
+              DvcOs = 'Windows',
+              DstAppType = "Process",
+              SrcUserIdType = iff (SrcUserId <> "S-1-0-0", "SID", ""),
+              SrcUserId = iff (SrcUserId <> "S-1-0-0", SrcUserId, ""),
+              DstUserIdType = iff (DstUserId <> "S-1-0-0", "SID", ""),
+              DstUserId = iff (DstUserId <> "S-1-0-0", DstUserId, ""),
+              SrcAppType = "Process",
+              EventType = "NetworkSession",
+              EventSchema = "NetworkSession",
+              EventSchemaVersion="0.2.3",
+              EventCount=toint(1),
+              EventVendor = "Microsoft",
+              EventProduct = "Windows Firewall",
+              EventStartTime = TimeGenerated,
+              EventEndTime = TimeGenerated,
+              EventSeverity = iff(EventID  in (5154, 5156, 5158), "Informational", "Low")
+      // -- Aliases
+      | extend 
+          Dvc = DvcHostname,
+          Hostname = DvcHostname,
+          IpAddr = SrcIpAddr,
+          Src = SrcIpAddr,
+          Dst = DstIpAddr,
+          Rule = tostring(NetworkRuleNumber),
+          DstDvcIdType = iff (DstDvcId != "", "SID", ""),
+          SrcDvcIdType = iff (SrcDvcId != "", "SID", "")
+      | lookup LayerCodeTable on LayerCode
+      | lookup ProtocolTable on Protocol
+      | project-away LayerCode, DirectionCode, Protocol, isOutBound, LayerName, EventID, LayerRTID,_ResourceId,_SubscriptionId
+      };
+      parser(starttime = datetime(null),endtime = datetime(null),srcipaddr_has_any_prefix = dynamic([]),dstipaddr_has_any_prefix = dynamic([]),ipaddr_has_any_prefix = dynamic([]),dstportnumber = int(null),hostname_has_any = dynamic([]),dvcaction = dynamic([]),eventresult = '*',disabled = false)
```

Comment:
please correct this.

parser(starttime = starttime , endtime = endtime.......

Reply:
done